packages/core/src/dispatcher.ts

```diff
@@ -97,27 +97,16 @@ export class Dispatcher {
         if (cacaoBlock) {
           const decodedProtectedHeader = base64urlToJSON(data.jws.signatures[0].protected)
           const capIPFSUri = decodedProtectedHeader.cap
-          const capCID = CID.parse(capIPFSUri.replace('ipfs://', ''))
-          const capFormat = await this._ipfs.codecs.getCodec(capCID.code).then((f) => f.name)
-          const capMhType = await this._ipfs.hashers
-            .getHasher(capCID.multihash.code)
-            .then((mh) => mh.name)
-          await this._ipfs.block.put(cacaoBlock, {
-            format: capFormat,
-            mhtype: capMhType,
-            version: capCID.version,
-          })
+          const putOptions = await Utils.getIPFSBlockPutOptionsFromCID(capIPFSUri, this._ipfs)
+          await this._ipfs.block.put(cacaoBlock, putOptions)
```

Comment:
I would just put the `_ipfs.block.put` part inside of the util function as well.